src/Service.Host/client/src/components/VatCode.js

```diff
@@ -0,0 +1,194 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from '@material-ui/core';
+import {
+    SaveBackCancelButtons,
+    InputField,
+    Loading,
+    Title,
+    ErrorCard
+} from '@linn-it/linn-form-components-library';
+import Page from '../containers/Page';
+
+function VatCode({
+    errorMessage,
+    editStatus,
+    history,
+    loading,
+    vatCodeId,
+    addVatCode,
+    setEditStatus,
+    updateVatCode,
+    ...props
+}) {
+    const [vatCode, setVatCode] = useState({});
+    const [prevVatCode, setPrevVatCode] = useState(null);
+
+    function handleSaveClick() {
+        updateVatCode(vatCodeId, vatCode);
+        setEditStatus('view');
+    }
+
+    function handleCancelClick() {
+        setVatCode(vatCode);
+        setEditStatus('view');
+    }
+
+    function handleAddClick() {
+        addVatCode(vatCode);
+        setEditStatus('view');
+    }
+
+    function handleBackClick() {
+        history.push('/products/maint/vat-codes');
+    }
+
+    function creating() {
+        return editStatus === 'create';
+    }
+
+    function editing() {
+        return editStatus === 'edit';
+    }
+
+    function codeInvalid() {
+        return !vatCode.code || vatCode.code.length === 0;
+    }
+
+    function descriptionInvalid() {
+        return !vatCode.description;
+    }
+
+    function rateInvalid() {
+        return typeof vatCode.rate !== 'number';
+    }
+
+    function inputInvalid() {
+        return codeInvalid() || descriptionInvalid() || rateInvalid();
+    }
+
+    function handleFieldChange(propertyName, newValue) {
+        setEditStatus('edit');
+        setVatCode({ ...vatCode, [propertyName]: newValue });
+    }
+
```

Comment:
This function could maybe be a candidate for being split out into a shared hook?